packages/@glimmer/node/lib/node-dom-helper.ts

```diff
@@ -11,16 +12,10 @@ export default class NodeDOMTreeConstruction extends DOMTreeConstruction {
   // override to prevent usage of `this.document` until after the constructor
   protected setupUselessElement() {}
 
-  insertHTMLBefore(parent: Simple.Element, reference: Simple.Node, html: string): Bounds {
-    let prev = reference ? reference.previousSibling : parent.lastChild;
-
+  insertHTMLBefore(parent: Simple.Element, reference: Option<Simple.Node>, html: string): Bounds {
     let raw = this.document.createRawHTMLSection(html);
     parent.insertBefore(raw, reference);
-
-    let first = prev ? prev.nextSibling : parent.firstChild;
-    let last = reference ? reference.previousSibling : parent.lastChild;
-
-    return new ConcreteBounds(parent, first, last);
```

Comment:
This is a bug (similar in nature to #852) in FastBoot. Since you don't normally run updates there I don't know how much this matters (i.e. should I split this into a BUGFIX PR and release that separately).